static/app/components/alert.tsx

```diff
@@ -19,50 +20,111 @@ export interface AlertProps extends React.HTMLAttributes<HTMLDivElement> {
 
 const DEFAULT_TYPE = 'info';
 
-const IconWrapper = styled('div')`
-  display: flex;
-  height: calc(${p => p.theme.fontSizeMedium} * ${p => p.theme.text.lineHeightBody});
-  margin-right: ${space(1)};
-  align-items: center;
-`;
+function Alert({
+  type = DEFAULT_TYPE,
+  showIcon = false,
+  opaque,
+  system,
+  expand,
+  trailingItems,
+  className,
+  children,
+  ...props
+}: AlertProps) {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const showExpand = defined(expand);
+  const showTrailingItems = defined(trailingItems);
+
+  // Show the hover state (with darker borders) only when hovering over the
+  // IconWrapper or MessageContainer.
+  const {hoverProps: iconHoverProps, isHovered: iconIsHovered} = useHover({
+    isDisabled: !showExpand,
+  });
+  const {hoverProps: messageHoverProps, isHovered: messageIsHovered} = useHover({
+    isDisabled: !showExpand,
+  });
+
+  function getIcon() {
+    switch (type) {
+      case 'warning':
+        return <IconWarning />;
+      case 'success':
+        return <IconCheckmark />;
+      case 'error':
+        return <IconNot />;
+      case 'info':
+      default:
+        return <IconInfo />;
+    }
+  }
 
-const ContentWrapper = styled('div')`
-  width: 100%;
-`;
+  function handleClick() {
+    showExpand && setIsExpanded(!isExpanded);
+  }
 
-const TrailingItems = styled('div')`
-  display: grid;
-  grid-auto-flow: column;
-  grid-template-rows: 100%;
-  align-items: center;
-  gap: ${space(1)};
-  height: calc(${p => p.theme.fontSizeMedium} * ${p => p.theme.text.lineHeightBody});
-  margin-left: ${space(1)};
-`;
+  return (
+    <Wrap
+      type={type}
+      system={system}
+      opaque={opaque}
+      expand={expand}
+      hovered={iconIsHovered || messageIsHovered}
+      className={classNames(type ? `ref-${type}` : '', className)}
+      {...props}
+    >
+      {showIcon && (
+        <IconWrapper onClick={handleClick} {...iconHoverProps}>
+          {getIcon()}
+        </IconWrapper>
+      )}
+      <ContentWrapper>
+        <MessageContainer
+          onClick={handleClick}
+          showIcon={showIcon}
+          showTrailingItems={showTrailingItems}
+          {...messageHoverProps}
+        >
+          <Message>{children}</Message>
+          {(showExpand || showTrailingItems) && (
+            <TrailingItemsWrap>
+              <TrailingItems onClick={e => e.stopPropagation()}>
```

Comment:
stop propagating `onClick` events that come from elements inside the `trailingItems` area, since that's where we put buttons and links.

Reply:
Should the expandIcon live inside the trailing items?

Reply:
yes, I want it to be aligned with the other trailing items, like so:
<img width="710" alt="Screen Shot 2022-04-11 at 10 16 08 AM" src="https://user-images.githubusercontent.com/44172267/162794404-8f3d0670-0688-40fe-87b0-63926b2da5ed.png">